tests/integration/src/test/java/org/netxms/tests/CollectorTest.java

```diff
@@ -0,0 +1,338 @@
+   public void testGetLastValues() throws Exception
+   {
+      final NXCSession session = connect();
+      session.syncObjects();
+
+      Collector testCollector = findOrCreateCollector(session, COLLECTOR_NAME);
+      assertNotNull(testCollector);
+
+      DciValue[] list = session.getLastValues(testCollector.getObjectId());
+      assertFalse(list.length > 0);
+
+      DataCollectionConfiguration dcc = session.openDataCollectionConfiguration(testCollector.getObjectId());
+
+      final DataCollectionItem dci = new DataCollectionItem(dcc, 0);
+
+      dci.setOrigin(DataOrigin.INTERNAL);
+      dci.setDescription(DCI_DESCRIPTION);
+      dci.setName(DUMMY);
+      dci.setPollingScheduleType(DataCollectionObject.POLLING_SCHEDULE_CUSTOM);
+      dci.setPollingInterval(Integer.toString(1));
+      dci.setTransformationScript("return 6;");
+
+      long id = dcc.modifyObject(dci);
+      Thread.sleep(1000);
+
+      list = session.getLastValues(testCollector.getObjectId());
+
+      assertTrue(list.length > 0); // Checking that DCI has been created
+
+      dcc.deleteObject(id);
+      dcc.close();
+      session.disconnect();
+   }
+
+   @Test
+   public void testThreshold() throws Exception
+   {
+      final NXCSession session = connect();
+      session.syncObjects();
+
+      Collector testCollector = findOrCreateCollector(session, COLLECTOR_NAME);
+      assertNotNull(testCollector);
+
+      DciValue[] list = session.getLastValues(testCollector.getObjectId());
+      assertFalse(list.length > 0);
+
+      DataCollectionConfiguration dcc = session.openDataCollectionConfiguration(testCollector.getObjectId());
+
+      final DataCollectionItem dci = new DataCollectionItem(dcc, 0);
+
+      dci.setOrigin(DataOrigin.INTERNAL);
+      dci.setDescription("test DCI for Collector");
+      dci.setName(DUMMY);
+      dci.setTransformationScript("return 6;");
+
+      Threshold threshold = new Threshold();
+      threshold.setValue("10");
+      threshold.setOperation(0);
+      dci.getThresholds().add(threshold);
+
+      long id = dcc.modifyObject(dci);
+      Thread.sleep(1000);
+
+      list = session.getLastValues(testCollector.getObjectId());
+      testCollector = findOrCreateCollector(session, COLLECTOR_NAME);
+
+      boolean findThreshold = false;
+      for(DciValue dciValue : list)
+      {
+         if (dciValue.getActiveThreshold() != null && testCollector.getStatus() != (ObjectStatus.NORMAL))
+            findThreshold = true;
+      }
+      assertTrue(findThreshold);
+
+      Alarm testAlarm = null;
+      HashMap<Long, Alarm> alarms = session.getAlarms();
+      for(Map.Entry<Long, Alarm> entry : alarms.entrySet())
+      {
+         Alarm alarm = entry.getValue();
+         if (alarm.getSourceObjectId() == testCollector.getObjectId() && alarm.getCurrentSeverity() != Severity.NORMAL)
+         {
+            testAlarm = alarm;
+            break;
+         }
+      }
+      assertNotNull(testAlarm);
+
+      dcc.deleteObject(id);
```

Comment:
I think we should extract this steps to postTest function to execute them even if test fails somwere in the middle